lib/session.js

```diff
@@ -2,45 +2,134 @@ import { query, update, sparqlEscapeUri } from 'mu';
 
 export async function getImpersonatedSession(sessionUri) {
   const response = await query(`
-  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
-  PREFIX muAccount: <http://mu.semte.ch/vocabularies/account/impersonation/>
-  SELECT
-    DISTINCT ?uri ?id ?impersonatedResource ?impersonatedResourceId
-  WHERE {
-    BIND(${sparqlEscapeUri(sessionUri)} AS ?uri)
-    ?uri mu:uuid ?id ;
-         muAccount:impersonates ?impersonatedResource .
-    ?impersonatedResource mu:uuid ?impersonatedResourceId .
-  }`);
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX muExt:  <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX muAccount: <http://mu.semte.ch/vocabularies/account/impersonation/>
+    PREFIX muSession: <http://mu.semte.ch/vocabularies/session/>
+
+    SELECT DISTINCT 
+      ?uri ?id ?impersonatedResource ?impersonatedResourceId ?originalResource ?originalResourceId ?originalSessionRoles ?originalSessionGroupId
+    WHERE {
+      BIND(${sparqlEscapeUri(sessionUri)} AS ?uri)
+      ?uri mu:uuid ?id ;
+        muSession:account ?impersonatedResource ;
+        muExt:originalResource ?originalResource ;
```

Comment:
Since we're in forked space now we can name this more specifically if we want: `muExt:originalAccount`?

Reply:
I agree with `muExt:originalAccount` as it feels "clearer" in a sense, especially since we use `muSession:account` to point to the actual session. But, we can wait for Felix to see what to go for.

Reply:
not really an opinion ATM. This service is going to spread over multiple stacks. how much more or less work will it be if we also decide to work towards a more generic version of this service?

Reply:
I think it wouldn't change much since the queries would need to be configurable anyways.